components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth2/device/response/DeviceFlowResponseTypeValidator.java

```diff
@@ -0,0 +1,49 @@
+/*
+ * Copyright (c) 2019, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+ *
+ * WSO2 Inc. licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.wso2.carbon.identity.oauth2.device.response;
+
+import org.apache.oltu.oauth2.common.OAuth;
+import org.apache.oltu.oauth2.common.error.OAuthError;
+import org.apache.oltu.oauth2.common.exception.OAuthProblemException;
+import org.apache.oltu.oauth2.common.validators.AbstractValidator;
+import org.wso2.carbon.identity.oauth2.device.constants.Constants;
+
+import javax.servlet.http.HttpServletRequest;
+
+/**
+ * Validator class for device response type.
+ */
+public class DeviceFlowResponseTypeValidator extends AbstractValidator<HttpServletRequest> {
+
+    public DeviceFlowResponseTypeValidator() {
+
+        this.requiredParams.add(Constants.RESPONSE_TYPE);
+        this.requiredParams.add(Constants.CLIENT_ID);
+    }
+
+    @Override
+    public void validateMethod(HttpServletRequest request) throws OAuthProblemException {
+
+        String method = request.getMethod();
+        if (!OAuth.HttpMethod.GET.equals(method) && !OAuth.HttpMethod.POST.equals(method)) {
```

Comment:
Is this mentioned in the specification?

Reply:
No this is used for response type validator. We are using authorize endpoint internally. For that purpose, I use this.